### DISCUSSION TITLE
Why the PnP name?

### DISCUSSION THREAD

**Discussion by BurtHarris** [pnp/powershell#1740] [Q&A]
Why do these packages use the name PnP?   On first reading, I assumed it stood for [Plug and Play](https://docs.microsoft.com/en-us/windows-hardware/drivers/kernel/introduction-to-plug-and-play), but that's pretty clearly not accurate.   I gather it relates to the [Microsoft Graph](https://docs.microsoft.com/en-us/azure/active-directory/develop/microsoft-graph-intro), APIs which **is** what I want.

---
Just to introduce myself, I'm a retired Microsoft SDE.   I've done a lot of automation work in PowerShell in the past.  These days I mostly just tinker.

**Comment by AndersRask** [comment]
I think the correct term was Plug and Pray ;-)

PnP in this context stands for Patterns & Practices, and have it's roots in the Patterns and Practices team from Microsoft back in the days.
This effort however, while condoned my Microsoft, is a community driven effort, but still strives to apply good patterns and practices on top of the sometimes wonky API's :-)
